pyproject.toml

```diff
@@ -33,6 +33,11 @@ dependencies = [
     'scooby>=0.5.1',
     "typing-extensions; python_version < '3.8'",
     'vtk',
+    'ipywidgets',
+    'trame>=2.2.6',
+    'trame-client>=2.4.2',
+    'trame-server>=2.8.0',
+    'trame-vtk>=2.0.17',
 ]
```

Comment:
The problem with this approach is that you end up increasing the needed dependencies to run PyVista. Do you really need ``ipywidgets`` or ``trame`` to run the typical use cases? The first one is mostly for Jupyter-related views. The same happens with ``trame`` AFAIK. They should really be optional dependencies, and the imports should be handled with caution by checking whether the module is available or not when doing the import statement (try/except approach).

But that's my gut feeling. Pinging @akaszynski @banesullivan and @AlejandroFernandezLuces for their thoughts.

Reply:
I totally agree. In fact, I was surprised to see those missing dependencies are in the optional dependencies requirements. Hence fixing this is more complex than I expected. There is something missing in the library structure.

Reply:
Something like this...

```py
try:
    import docker
    from docker.errors import APIError, ContainerError, ImageNotFound
    from docker.models.containers import Container


    _HAS_DOCKER = True
except ModuleNotFoundError:  # pragma: no cover
    _HAS_DOCKER = False

```

Reply:
But of course... applied to the modules involved in this issue 😄 not ``docker``

Reply:
We actually want to remove `pyvista.trame` from the top level namespace of the package to avoid this issue rather than adding additional imports or "has package" logic.

See https://github.com/pyvista/pyvista/pull/3943

Reply:
Sounds good to me @banesullivan!